getting-started/deploying-to-openshift.adoc

```diff
@@ -0,0 +1,235 @@
+include::../partials/attributes.adoc[]
+
+This guide provides instructions for deploying a business application to OpenShift using three different methods. Ensure that the following prerequisites are in place before proceeding.
+
+== Prerequisites
+
+* https://docs.openshift.com/container-platform/4.16/cli_reference/openshift_cli/getting-started-cli.html#installing-openshift-cli[OpenShift CLI (`oc`)]
+* https://docs.docker.com/manuals/[Docker]
+
+== Deployment Options
+
+=== 1. Build and Deploy on OpenShift
+
+Use this method to build and deploy the application directly on OpenShift.
+
+[.steps]
+
+. **Log in to OpenShift CLI:**
++
+[source,shell]
+----
+oc login
+----
+
+. **Add the OpenShift extension to the Quarkus project:**
++
+[source,shell]
+----
+mvn quarkus:add-extension -Dextensions='quarkus-openshift'
+----
+
+. **Update the `src/main/resources/application.properties` file to enable the creation of an OpenShift Route:**
++
+[source,properties]
+----
+# OpenShift
+quarkus.openshift.route.expose=true
+## Route TLS configuration:
+quarkus.openshift.route.tls.termination=edge
+quarkus.openshift.route.tls.insecure-edge-termination-policy=None
+----
+
+. **Build and deploy the application:**
++
+[source,shell]
+----
+mvn install -Dquarkus.openshift.deploy=true
+----
+
+=== 2. Build Locally with Docker and Automatically Deploy on OpenShift
+
+This method involves building the container image for the application locally using Docker, pushing the image to registry, followed by an automatic deployment to OpenShift.
+
+[.steps]
+
+. **Log in to OpenShift CLI:**
++
+[source,shell]
+----
+oc login
+----
+
+. **Add the OpenShift extension to the Quarkus project:**
++
+[source,shell]
+----
+mvn quarkus:add-extension -Dextensions='quarkus-openshift'
+----
+
+. **Update the `src/main/resources/application.properties` file with the following properties to configure the image and enable the creation of an OpenShift Route:**
++
+[source,properties]
+----
+# Container Image Builder
+quarkus.container-image.builder=docker
+quarkus.container-image.registry=<image_registry>
+quarkus.container-image.group=<registry_group_or_user>
+quarkus.container-image.name=<image_name>
+quarkus.container-image.tag=<image_tag>
+quarkus.docker.buildx.platform=linux/amd64
+# OpenShift
+quarkus.openshift.route.expose=true
+## Route TLS configuration:
+quarkus.openshift.route.tls.termination=edge
+quarkus.openshift.route.tls.insecure-edge-termination-policy=None
+----
+
+[NOTE]
+====
+* `<image_registry>`: The URL of the container image registry where the Docker image will be stored. Example: `docker.io`.
+* `<registry_group_or_user>`: The group or user in the container image registry. Example: `myuser`.
+* `<image_name>`: The name of the Docker image to be built. Example: `myapp`.
+* `<image_tag>`: The tag of the Docker image. Example: `v1.0.0`.
+
+Replace these tokens with appropriate values based on the environment setup.
+====
```

Comment:
I'd recommend using callouts instead of notes: https://docs.asciidoctor.org/asciidoc/latest/verbatim/callouts/

Reply:
I've added it! Hopefully, the callout icons won't be selected or copied when the user tries to copy and paste the code blocks.

Reply:
They are not. I believe they're added via a CSS trick which is not actually part of the rendered text.

Reply:
At least, they shouldn’t be copied :)